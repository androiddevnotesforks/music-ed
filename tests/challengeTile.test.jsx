import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import ChallengeTile from '../components/challengeTile'

const mockProps = {
  challenge: {
    name: "Challenge Name",
    info: "Challenge Info",
    key: "Challenge Key",
    numberQs: 24,
    range: "C4-C5",
    intervals: [2, 3],
    direction: "Ascending",
    exLink: "/exLink"
  },
  path: "/path"
}

jest.mock("next/link", () => {
  return ({children}) => {
      return children;
  }
}); 

// Link not tested

describe("ChallengeTile component", () => {
  beforeEach(() => {
    render(<ChallengeTile {...mockProps}/>)
  })
  it('renders', () => {
    const challengeTile = screen.getByTestId("challenge-tile-div")
    expect(challengeTile).toBeInTheDocument()
  })
  it("Matches snapshot", () => {
    const challengeTile = screen.getByTestId("challenge-tile-div")
    expect(challengeTile).toMatchSnapshot()
  })
  it("Challenge name stated", () => {
    const challengeTile = screen.getByTestId("challenge-tile-div")
    expect(challengeTile.textContent).toMatch(mockProps.challenge.name)
  })
  it("Challenge info stated", () => {
    const challengeTile = screen.getByTestId("challenge-tile-div")
    expect(challengeTile.textContent).toMatch(mockProps.challenge.info)
  })

})