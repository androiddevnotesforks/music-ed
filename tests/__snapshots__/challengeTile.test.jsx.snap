// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChallengeTile component Matches snapshot 1`] = `
<div
  class="w-full m-2 p-2 border shadow lg:w-1/2 "
  data-testid="challenge-tile-div"
>
  <h2
    class="font-semibold"
  >
    Challenge Name
  </h2>
  <p
    class="font-thin"
  >
    Challenge Info
  </p>
</div>
`;
